# Réseau cristallin du MgI2 
Ce code a pour but d'analyser **la structure** du cristal **MgI2** (mp-23205) à partir des données fournies sur site du Materials Project [1].

Le fichier exploité ci-dessous est le fichier CIF du MgI2 [2] téléchargé sur le site du Materials Project.

La reflexion théorique sous-jacente à ce code est inspirée du cours *LMAPR1492 - Physique des matériaux*, donné par le professeur Gian-Marco RIGNANESE [3].


In [38]:
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.ext.matproj import MPRester
with MPRester("ktopgq8J6C3OwUSa") as m:
    structure = m.get_structure_by_material_id("mp-23205")
    

Pour caractériser un réseau cristallin, il suffit de caractériser **la maille primitive**. 

C'est le motif de base qui, en se répetant, permet de reconstituer entièrement le réseau.

In [39]:
    #Takes a pymatgen.core.structure.Structure object and a symprec. 
    #Uses spglib to perform various symmetry finding operations.
    sa = SpacegroupAnalyzer(structure)

    #Gives a structure with a primitive cell according to certain standards
    prim = SpacegroupAnalyzer(structure).get_primitive_standard_structure()

# 1. Système cristallin

Les systèmes cristallins sont des manière d'agencer périodiquement les points du réseau. 

In [40]:
    #Get the crystal system for the structure, e.g., (triclinic, orthorhombic, cubic, etc.).
    systeme = sa.get_crystal_system()
    print('Système cristallin : ' + systeme)
    

Système cristallin : trigonal


Le système trigonal ou rhombohédrique est caractérisé par :
* $a = b = c$
* $\alpha = \beta = \gamma \ne 90^\circ$

# 2. Type de maille
La **maille (primitive)** ou **cellule (unité)** est une (la plus petite) partie du réseau, *un (le) motif de base*,  qui, en se répettant par translation, permet de la reconstituer.


In [41]:
    #Get the lattice for the structure, e.g., (triclinic, orthorhombic, cubic, etc.).
    #This is the same than the crystal system with the exception of the hexagonal/rhombohedral lattice
    lattice = sa.get_lattice_type()
    print('type de maille : ' + lattice)

type de maille : hexagonal


Dans le cas du MgI2, la maille est primitive: on a donc un système triogonal P (ou hexagonal)

Les réseaux de Bravais  sont caractérisé par le système cristallin et le type de maille
La maille hexagonale ou trigonale P est caractérisé par :
* $a = b \ne c$
* $\alpha = \beta = 90^\circ$
* $\gamma = 120^\circ$

# 3. Groupe ponctuel
L'ensemble des opérations ponctuelles de symétrie forment un groupe ponctuel (G) au sens mathématique du terme.

Il y a 5 groupes ponctuels avec un système trigonal : 
$$3,\overline{3}, 32, 3m, \overline{3}m$$

In [45]:
    reciproque = sa.get_point_group_symbol()
    print("groupe ponctuel : " + reciproque)

groupe ponctuel : -3m


Étant donné le système trigonal P et le groupe ponctuel obtenu ($\overline{3}m$), nous pouvons dire que :
* L'axe c est d'ordre 3
* L'axe c est unique
* le système contient un plan miroir le long de l'axe c

# 4. Vecteurs de base du réseau direct
L'empilement ordonné est appelé réseau ponctuel (ou direct). Pour une structure donnée, le réseau est défini en se donnant un système d'axes et en répetant toutes les positions identiques par rapport à l'origine.

Le réseau est défini par des vecteurs de base **a, b, c** indépendants et tels que tout point du réseau direct peut s'écrire :
$${\bf R} = l {\bf a} + m {\bf b} + n {\bf c}$$

In [43]:
    vector_direct = prim.lattice
    print (vector_direct)
    print('----------------------------')
    print('a =' + str(vector_direct.a))
    print('b =' + str(vector_direct.b))
    print('c =' + str(vector_direct.c))
    print('alpha =' + str(vector_direct.alpha))
    print('beta =' + str(vector_direct.beta))
    print('gamma =' + str(vector_direct.gamma))


2.104749 -3.645532 0.000000
2.104749 3.645532 0.000000
0.000000 0.000000 7.731276
----------------------------
a =4.2094982175295
b =4.2094982175295
c =7.73127647
alpha =90.0
beta =90.0
gamma =120.00000000000001


Les vecteurs de base du réseau direct sont donc :
* ${\bf a} = [2.104749; -3.645532; 0.000000]$
* ${\bf b} = [2.104749; 3.645532; 0.000000]$
* ${\bf c} = [0.000000; 0.000000; 7.731276]$

Les paramètres de mailles obtenus confirmes bien les conclusions du point 2 : 
*  $a = b \ne c$
    * Longueur de a = 4.2094982175295 Å 
    * Longueur de b = 4.2094982175295 Å 
    * Longueur de c = 7.73127647 Å
* $\alpha = \beta = 90^\circ$, $\gamma = 120^\circ$
    * $\alpha = 90.0^\circ$
    * $\beta = 90.0^\circ$
    * $\gamma = 120.00000000000001^\circ$


# 5. Vecteurs de base du réseau réciproque
L'ensemble de tous les vecteurs d'onde **K** qui sont associés à des ondes planes qui ont la même périodicité  que le réseau de départ (réseau direct) forment égalament un réseau appelé **réseau réciproque**.

L'ensemble des vecteurs d'onde **K** vérifient la relation :
$$e^{i{\bf K R}}=1$$

In [44]:
    vector_reciprocal = vector_direct.reciprocal_lattice
    print (vector_reciprocal)
    print('----------------------------')
    print('a* =' + str(vector_reciprocal.a))
    print('b* =' + str(vector_reciprocal.b))
    print('c* =' + str(vector_reciprocal.c))
    print('alpha* =' + str(vector_reciprocal.alpha))
    print('beta* =' + str(vector_reciprocal.beta))
    print('gamma* =' + str(vector_reciprocal.gamma))

1.492621 -0.861765 0.000000
1.492621 0.861765 0.000000
0.000000 0.000000 0.812697
----------------------------
a* =1.7235302361512468
b* =1.7235302361512466
c* =0.8126970147246055
alpha* =90.0
beta* =90.0
gamma* =59.99999999999999


Les vecteurs de base du réseau direct sont donc :
* ${\bf a^*} = [1.492621; -0.861765; 0.000000]$
* ${\bf b^*} = [1.492621; 0.861765; 0.000000]$
* ${\bf c^*} = [0,000000; 0,000000; 0.812697$

Les paramètres de mailles obtenus confirmes encore bien les conclusions faites précédemment: 
*  $a^* = b^* \ne c^*$
    * Longueur de a* = 1.7235302361512468 Å 
    * Longueur de b* = 1.7235302361512468 Å 
    * Longueur de c* = 0.8126970147246055 Å
* $\alpha^* = \beta^* = 90^\circ$, $\gamma^* = \frac{120^\circ}{2}$
    * $\alpha^* = 90.0^\circ$
    * $\beta^* = 90.0^\circ$
    * $\gamma^* = 59.99999999999999^\circ$

# Bibliographie
[1] *Materials project mp-23205*, URL: https://materialsproject.org/materials/mp-23205/

[2] *MgI2_mp-23205_symmetrized.cif*, URL: https://github.com/laralej/lmapr1492/blob/master/MgI2_mp-23205_symmetrized.cif

[3] *Chapitre 1 : Cristallographie géométrique*, URL: https://moodleucl.uclouvain.be/pluginfile.php/1274322/mod_resource/content/3/Chap-1.pdf 